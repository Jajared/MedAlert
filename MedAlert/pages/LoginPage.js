import { SafeAreaView, Text, StyleSheet, View, TextInput, TouchableOpacity, StatusBar, Image } from "react-native";
import { useState } from "react";
import { Feather } from "@expo/vector-icons";
import { AntDesign } from "@expo/vector-icons";
import { LinearGradient } from "expo-linear-gradient";

export default function LoginPage({ navigation, onLogin }) {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [showPassword, setShowPassword] = useState(false);
  const togglePasswordVisibility = () => {
    setShowPassword(!showPassword);
  };

  return (
    <SafeAreaView style={styles.container}>
      <StatusBar barStyle="dark-content" />
      <Image source={require("../assets/medalert_icon.png")} style={styles.logo}></Image>
      <Text style={styles.title}>Welcome! ðŸ‘‹</Text>
      <View style={styles.inputBox}>
        <AntDesign name="mail" size={20} color="black" style={styles.inputIcon} />
        <TextInput style={styles.inputText} value={email} placeholder="Email" onChangeText={(text) => setEmail(text)}></TextInput>
      </View>
      <View style={styles.inputBox}>
        <AntDesign name="lock" size={26} color="black" style={styles.inputIcon} />
        <TextInput style={{ flex: 8 }} value={password} secureTextEntry={!showPassword} placeholder="Password" onChangeText={(text) => setPassword(text)}></TextInput>
        <TouchableOpacity style={styles.showButton} onPress={togglePasswordVisibility}>
          <Feather name={showPassword ? "eye-off" : "eye"} size={20} color="gray" />
        </TouchableOpacity>
      </View>
      <View style={styles.optionsBar}>
        <TouchableOpacity onPress={() => navigation.navigate("Reset Password")}>
          <Text style={{ color: "#FF014E" }}>Forgot Password?</Text>
        </TouchableOpacity>
      </View>

      <TouchableOpacity
        onPress={async () => {
          try {
            const success = await onLogin(email, password);
            if (success) {
              navigation.navigate("Home");
            } else {
              alert("Please retry logging in");
            }
          } catch (error) {
            console.log(error);
          }
        }}
        style={styles.buttonContainer}
      >
        <LinearGradient colors={["#FFA7AF", "#FF014E"]} style={styles.gradient}>
          <Text style={styles.buttonText}>Login</Text>
        </LinearGradient>
      </TouchableOpacity>
      <View style={styles.emptySection}></View>
      <View style={styles.signUpSection}>
        <Text style={{ fontSize: 15 }}>Don't have any account?</Text>
        <TouchableOpacity onPress={() => navigation.navigate("Sign Up Home")}>
          <Text style={{ color: "#FF014E", fontSize: 15 }}> Sign Up</Text>
        </TouchableOpacity>
      </View>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "column",
    backgroundColor: "#fff",
    alignItems: "center",
    justifyContent: "center",
  },
  logo: {
    height: 100,
    width: 100,
    marginVertical: 60,
  },
  title: {
    flex: 1,
    fontSize: 25,
    fontWeight: "bold",
    alignSelf: "flex-start",
    marginLeft: 20,
  },
  optionsBar: {
    flex: 1,
    flexDirection: "row",
    justifyContent: "flex-end",
    width: "90%",
    margin: 10,
  },
  inputIcon: {
    flex: 1,
    alignSelf: "center",
  },
  inputBox: {
    flex: 1,
    flexDirection: "row",
    borderBottomColor: "grey",
    borderBottomWidth: 1,
    borderRadius: 10,
    padding: 10,
    color: "grey",
    margin: 10,
  },
  inputText: {
    flex: 9,
    alignSelf: "center",
    color: "grey",
  },
  showButton: {
    flex: 1,
    alignSelf: "center",
  },
  buttonContainer: {
    flex: 2,
    margin: 10,
    width: "90%",
  },
  gradient: {
    paddingVertical: 15,
    paddingHorizontal: 100,
    borderRadius: 10,
  },
  buttonText: {
    color: "#fff",
    fontSize: 16,
    fontWeight: "bold",
    alignSelf: "center",
  },
  emptySection: {
    flex: 4,
  },
  signUpSection: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    flexDirection: "row",
  },
});
