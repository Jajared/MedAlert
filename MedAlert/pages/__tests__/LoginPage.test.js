import { render, screen, fireEvent } from "@testing-library/react-native";
import LoginPage from "../LoginPage";
import { expect, jest, test } from "@jest/globals";

jest.mock("expo-font");
jest.mock("expo-asset");

const navigationObject = {
  navigate: jest.fn(),
  goBack: jest.fn(),
};

describe("AboutPage", () => {
  // Check if renders correctly
  it("renders correctly", () => {
    const tree = render(<LoginPage navigation={navigationObject} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  // Check number of children
  it("has 9 children", () => {
    render(<LoginPage navigation={navigationObject} />);
    expect(screen.getByTestId("LoginPage").children.length).toBe(9);
  });
  // Forget password button is pressed
  it("Forget password button is pressed", () => {
    render(<LoginPage navigation={navigationObject} />);
    fireEvent.press(screen.getByTestId("ForgetButton"));
    expect(navigationObject.navigate).toHaveBeenCalled();
  });
  // Sign up button is pressed
  it("Sign up button is pressed", () => {
    render(<LoginPage navigation={navigationObject} />);
    fireEvent.press(screen.getByTestId("SignUpButton"));
    expect(navigationObject.navigate).toHaveBeenCalled();
  });
});
