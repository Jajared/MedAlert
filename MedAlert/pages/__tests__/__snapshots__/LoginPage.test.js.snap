// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LoginPage/> renders correctly 1`] = `
<RCTSafeAreaView
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Image
    source={1}
    style={
      {
        "flex": 7,
        "marginVertical": 20,
        "resizeMode": "contain",
      }
    }
  />
  <Text
    style={
      {
        "alignSelf": "flex-start",
        "flex": 1,
        "fontSize": 25,
        "fontWeight": "bold",
        "marginLeft": 20,
      }
    }
  >
    Welcome! ðŸ‘‹
  </Text>
  <View
    style={
      {
        "borderBottomColor": "grey",
        "borderBottomWidth": 1,
        "borderRadius": 10,
        "color": "grey",
        "flex": 1,
        "flexDirection": "row",
        "margin": 10,
        "padding": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "black",
            "fontSize": 20,
          },
          {
            "alignSelf": "center",
            "flex": 1,
          },
          {
            "fontFamily": "anticon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      î™™
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder="Email"
      style={
        {
          "alignSelf": "center",
          "color": "grey",
          "flex": 9,
        }
      }
      value=""
    />
  </View>
  <View
    style={
      {
        "borderBottomColor": "grey",
        "borderBottomWidth": 1,
        "borderRadius": 10,
        "color": "grey",
        "flex": 1,
        "flexDirection": "row",
        "margin": 10,
        "padding": 10,
      }
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "black",
            "fontSize": 26,
          },
          {
            "alignSelf": "center",
            "flex": 1,
          },
          {
            "fontFamily": "anticon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      î™»
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder="Password"
      secureTextEntry={true}
      style={
        {
          "flex": 8,
        }
      }
      value=""
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignSelf": "center",
          "flex": 1,
          "opacity": 1,
        }
      }
    >
      <Text
        allowFontScaling={false}
        selectable={false}
        style={
          [
            {
              "color": "gray",
              "fontSize": 20,
            },
            undefined,
            {
              "fontFamily": "feather",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        ï…£
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-end",
        "margin": 10,
        "width": "90%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "#FF014E",
          }
        }
      >
        Forgot Password?
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "flex": 2,
        "margin": 10,
        "opacity": 1,
        "width": "90%",
      }
    }
  >
    <ViewManagerAdapter_ExpoLinearGradient
      proxiedProperties={
        {
          "colors": [
            4294944687,
            4294902094,
          ],
          "endPoint": undefined,
          "locations": undefined,
          "startPoint": undefined,
        }
      }
      style={
        {
          "borderRadius": 10,
          "paddingHorizontal": 100,
          "paddingVertical": 15,
        }
      }
    >
      <Text
        style={
          {
            "alignSelf": "center",
            "color": "#fff",
            "fontSize": 16,
            "fontWeight": "bold",
          }
        }
      >
        Login
      </Text>
    </ViewManagerAdapter_ExpoLinearGradient>
  </View>
  <View
    style={
      {
        "flex": 4,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 15,
        }
      }
    >
      Don't have any account?
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "color": "#FF014E",
            "fontSize": 15,
          }
        }
      >
         Sign Up
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
